### DISCUSSION TITLE
How to publish using GPU mode in python

### DISCUSSION THREAD

**Discussion by boysj** [ultralytics/ultralytics#13775] [Q&A]
If I export it as an onnx model, the published file will be as high as 1G+. If you use the default yolov library, it will still be very large after release.
Does anyone know how to solve it?

**Comment by glenn-jocher** [comment]
@boysj hi there! 👋

To publish your model using GPU mode in Python while keeping the file size manageable, you can consider the following steps:

1. **Model Pruning and Quantization**:
   - **Pruning**: Remove unnecessary parts of the model to reduce its size.
   - **Quantization**: Convert the model to a lower precision (e.g., FP16 or INT8) to reduce size and improve inference speed.

2. **Export to TensorRT**:
   TensorRT is highly optimized for NVIDIA GPUs and can significantly reduce model size and improve inference speed. Here's how you can export your model to TensorRT:

   ```python
   from ultralytics import YOLO

   # Load a YOLOv8 model
   model = YOLO("yolov8n.pt")

   # Export the model to TensorRT format
   model.export(format="engine")  # This creates 'yolov8n.engine'

   # Load the exported TensorRT model
   trt_model = YOLO("yolov8n.engine")

   # Run inference
   results = trt_model("https://ultralytics.com/images/bus.jpg")
   ```

3. **Ensure Latest Versions**:
   Make sure you are using the latest versions of `torch` and `ultralytics`. You can upgrade them using:
   ```bash
   pip install --upgrade torch ultralytics
   ```

4. **Optimize ONNX Model**:
   If you still prefer using ONNX, you can optimize the ONNX model using tools like `onnx-simplifier` and `onnxruntime`.

   ```bash
   pip install onnx-simplifier
   python3 -m onnxsim yolov8n.onnx yolov8n_simplified.onnx
   ```

If you encounter any issues during these steps, please provide a minimum reproducible example so we can assist you further. You can find more details on creating a reproducible example [here](https://docs.ultralytics.com/help/minimum_reproducible_example).

I hope this helps! Let us know if you have any other questions. 😊
